components/Banner/examples/BannerTypicalUseExampleViewController.m

```diff
@@ -129,230 +130,238 @@ + (NSDictionary *)catalogMetadata {
-    [self.bannerViewContainer removeFromSuperview];
+  if (self.bannerView) {
+    [self.bannerView removeFromSuperview];
   }
-  // Set up banner view container
-  UIView *bannerViewContainer = [[UIView alloc] initWithFrame:CGRectZero];
-  [self.view addSubview:bannerViewContainer];
-  bannerViewContainer.backgroundColor = [UIColor whiteColor];
-  self.bannerViewContainer = bannerViewContainer;
 
   MDCBannerView *bannerView = [[MDCBannerView alloc] init];
-  bannerView.text = exampleLongText;
-  bannerView.image = [UIImage imageNamed:@"Email"];
-  bannerView.numberOfButtons = 2;
-
-  UIFont *buttonFont = [MDCTypography body2Font];
-  MDCButton *dismissButton = bannerView.buttons[0];
-  [dismissButton setTitle:@"DISMISS" forState:UIControlStateNormal];
-  [dismissButton setTitleFont:buttonFont forState:UIControlStateNormal];
-  [dismissButton setTitleFont:buttonFont forState:UIControlStateHighlighted];
+  bannerView.textLabel.text = exampleLongText;
+  bannerView.backgroundColor = self.colorScheme.surfaceColor;
+  UIEdgeInsets margins = UIEdgeInsetsZero;
+  margins.left = exampleBannerContentPadding;
+  margins.right = exampleBannerContentPadding;
+  bannerView.layoutMargins = margins;
+  [self.view addSubview:bannerView];
+  self.bannerView = bannerView;
+
+  MDCButton *dismissButton = bannerView.leadingButton;
+  [dismissButton setTitle:@"Dismiss" forState:UIControlStateNormal];
+  dismissButton.uppercaseTitle = YES;
   [dismissButton setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   dismissButton.backgroundColor = self.colorScheme.surfaceColor;
   [dismissButton sizeToFit];
   [dismissButton addTarget:self
                     action:@selector(dismissBanner)
           forControlEvents:UIControlEventTouchUpInside];
-  MDCButton *changeTextButton = bannerView.buttons[1];
-  [changeTextButton setTitle:@"CHANGE TEXT" forState:UIControlStateNormal];
-  [changeTextButton setTitleFont:buttonFont forState:UIControlStateNormal];
-  [changeTextButton setTitleFont:buttonFont forState:UIControlStateHighlighted];
+  MDCButton *changeTextButton = bannerView.trailingButton;
+  [changeTextButton setTitle:@"Long dismiss" forState:UIControlStateNormal];
+  changeTextButton.uppercaseTitle = YES;
   [changeTextButton setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   changeTextButton.backgroundColor = self.colorScheme.surfaceColor;
-  [changeTextButton sizeToFit];
   [changeTextButton addTarget:self
-                       action:@selector(changeText)
+                       action:@selector(dismissBanner)
              forControlEvents:UIControlEventTouchUpInside];
 
-  self.bannerView = bannerView;
-  [self.bannerViewContainer addSubview:bannerView];
-  CGSize bannerViewSize =
-      [bannerView sizeThatFits:CGSizeMake(exampleBannerContentMultiLineWidth, CGFLOAT_MAX)];
-  bannerView.frame = CGRectMake(0.0f, 0.0f, bannerViewSize.width, bannerViewSize.height);
-
   // Adjust bannerViewContainer's frame
```

Comment:
Why do you have to adjust the banner's frame? Isn't this handled in `viewDidLayoutSubviews`?

Reply:
Good point! It was necessary before because frame needs to be set before `viewDidLayoutSubviews`. Some changes made in 3972bb8b4d40b8882b6275e31863f5da28be00f6 to address it and removed some duplicate code blocks. It's cleaner now.